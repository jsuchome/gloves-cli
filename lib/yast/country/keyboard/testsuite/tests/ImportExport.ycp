/**
 * tests for Keyboard.ycp autoyast functions
 * Author:	jsuchome@suse.cz
 * $Id: ImportExport.ycp 57028 2009-04-29 10:58:09Z lslezak $
 */
{
    // testedfiles: Keyboard.ycp

    import "Testsuite";

    map READ = $[
	"target" : $[
	    "size"		: 1,
	    "dir"		: [],
	    "yast2"		: $[
		"czech-qwerty"	: [
		    "Czech (qwerty)",
		    $[
			"pc104": $[
			    "ncurses": "cz-lat2-us.map.gz",
			    "compose" : "latin2"
			],
		    ]
		],
	    ],
	],
	"probe"	: $[
	    "architecture"	: "i386"
	],
    ];

    map E	= $[
	"target" : $[
	    "bash_output"	: $[],
	],
    ];

    import "Mode";

    // let's simulate Mode::config
    Mode::SetMode ("autoinst_config");

    Testsuite::Init ([READ,$[],E], nil);

    import "Keyboard";

    Testsuite::Test (``(Keyboard::Modified ()), [], nil);

    map imported	= $[
	"keymap"	: "czech-qwerty"
    ];

    Testsuite::Test (``(Keyboard::Import (imported)), [READ,$[],E], nil);

    Testsuite::Test (``(Keyboard::Modified ()), [], nil);

    Testsuite::Test (``(Keyboard::Export ()), [], nil);

    Testsuite::Test (``(Keyboard::Summary ()), [], nil);

}
