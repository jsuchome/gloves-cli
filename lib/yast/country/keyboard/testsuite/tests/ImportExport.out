Return	false
Read	.target.yast2 "keyboard_raw.ycp" $["czech-qwerty":["Czech (qwerty)", $["pc104":$["compose":"latin2", "ncurses":"cz-lat2-us.map.gz"]]]]
Return	true
Return	true
Return	$["keymap":"czech-qwerty"]
Return	<ul><li>Current Keyboard Layout: Czech (qwerty)</li></ul>
